dbms/src/Storages/Page/V3/BlobStore.cpp

```diff
@@ -251,9 +251,17 @@ void BlobStore::remove(const PageEntriesV3 & del_entries)
         // So if we can't use id find blob, just ignore it.
         if (stat)
         {
-            LOG_FMT_TRACE(log, "Blob begin to recalculate capability [blob_id={}]", blob_id);
-            auto lock = stat->lock();
-            stat->recalculateCapacity();
+            {
+                auto lock = stat->lock();
+                stat->recalculateCapacity();
+            }
+            LOG_FMT_TRACE(log, "Blob recalculated capability [blob_id={}] [max_cap={}] "
```

Comment:
Why move this log out of the lock scope? I think if you write `stat` under the protection of lock, you should also read it under the protection of lock.

Reply:
These logs are only used to inform the blob situation after MVCC GC, we do not need an accurate value to count it. So I don't think that we need to put these rows into `lock`, otherwise, it will be slower.

Reply:
OK